src/xalloc.c

```diff
@@ -158,7 +378,7 @@ void *xcalloc(size_t n, size_t size)
         return NULL;
     }
 
-    insert_ptr(p);
+    insert_ptr_legacy(p);
     return p;
 }
 

```

Comment:
You can replace the whole of the contents of xrealloc with 'return arena_xrealloc(&global_arena, old_ptr, new_size);